rgo/git.py

```diff
@@ -149,6 +149,52 @@ def describe(self, pkg=None):
         version = version.replace("-", "_")
         return version, release
 
+    def _check_output(self, cmd):
+        env = dict(os.environ)
+        env["LC_ALL"] = "C.UTF-8"
+        output = subprocess.check_output(cmd, cwd=self.cwd, encoding="utf-8", env=env).strip().splitlines()
+        return output
+
+    def get_describe_long(self, ref):
+        """
+        Run git describe --tags --long <ref>
+        Return the a tuple with the following data:
+          * desc: raw `git desribe --tags --long <ref>` output
+          * tag: last available tag in commits <= ref (parsed from desc)
+          * commits: number of commits between the tag and the ref (parsed from desc)
+          * hash: hash of the ref (parsed from desc)
+        """
+        cmd = ["git", "describe", "--tags", "--long", ref]
+        desc = self._check_output(cmd)[0].strip()
+
+        tag, commits, hash = desc.rsplit("-", 2)
+
+        # covert number of commits to integer
+        commits = int(commits)
+
+        # remove the 'g' prefix and keep only the hash
+        hash = hash[1:]
+
+        return desc, tag, commits, hash
+
+    def get_rpm_changelog(self):
+        """
+        Generate RPM changelog records from git history between self.ref and the last tag.
+        If self.ref == last tag, generate a changelog between self.ref and the last but one tag.
+        """
+        ref_desc, ref_tag, ref_commits, ref_hash = self.get_describe_long(self.ref)
```

Comment:
A lot of unused variables. I question the overall usefulness of `ref_desc`, the other ones might be potentially useful... maybe use `_` or `unused` for those? (I've configured PEP8 Speaks for the repo, not sure if it picks this up, we'll see after you force-push :slightly_smiling_face:)